styleguide/REACT.md

```diff
@@ -427,6 +427,22 @@
   export type { Props as ButtonProps };
   ```
 
+- Event handler props should use `on`, while component methods used as event handlers should use `handle` as prefix followed by the event name. eslint: [`react/jsx-handler-names`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-handler-names.md)
```

Comment:
@dricholm This looks good to me!

I spent a little time this afternoon pulling out shared `react` and `typescript-eslint` rules into this repo. That way, instead of going to `z-web` and `z-web-angular` to set up this rule, you can set it up as part of this PR in the `react.js` file.

Please rebase this PR, and add the config to the `react.js` file for `react/jsx-handler-names`. Once you've merged this PR, I'll set you up with access to our `npm` organisation so that you can publish a new version of this package!

Reply:
After trying out the eslint rule it doesn't work as I expected it would. It doesn't seem to enforce the `on` prefix for props. Also calling `setState()` in an inline handler function will also trigger an eslint error even though `checkInlineFunction` is `false`.

So if it is okay, I would just add the point about using `on` as prefix for event handler props, but without an eslint rule to enforce it.
What do you think?

Reply:
@dricholm I think that's totally fine! Be sure to remove the reference to the rule in the description.